peps/pep-0791.rst

```diff
@@ -0,0 +1,158 @@
+module, called ``imath``:
+
+* :external+py3.14:func:`~math.comb`
+* :external+py3.14:func:`~math.factorial`
+* :external+py3.14:func:`~math.gcd`
+* :external+py3.14:func:`~math.isqrt`
+* :external+py3.14:func:`~math.lcm`
+* :external+py3.14:func:`~math.perm`
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Modules functions will accept integers and objects that implement the
+:external+py3.14:meth:`~object.__index__` method which is used to convert the
+object to an integer number.
+
+Possible extensions for the new module and its scope are discussed in the
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for an indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+https://github.com/python/cpython/pull/133909
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Chosen name seems consistent with other domain-specific mathematical module:
+:external+py3.14:mod:`cmath` (for complex numbers).
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one four years ago.  Its repository is no longer accessible.
+The `Imath <https://github.com/AcademySoftwareFoundation/Imath>`_ C++ library
+include Python bindings with the same name.
+
+`Polling shows <https://discuss.python.org/t/91337/35>`_ ``intmath`` as another
+popular name.  The argument made was that the normal mathematical spelling of
+the imaginary unit is ``i``, which makes imath ambiguous.  It also has no conflict
+with any PyPI module.  On the other hand, ``intmath`` may be confused with
+interval math or numerical integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
+``integermath`` and ``imaths``.
+
+
+Module scope and possible extensions
+------------------------------------
+
+Unless we can just provide bindings to some well supported mathematical library
+like the GMP, the module scope should be limited.  For example, no efficient primality
+testing and factorization.
+
+There are possible additions, among proposed in the initial discussion thread
+(see also issue
+`python/cpython#81313 <https://github.com/python/cpython/issues/81313>`_):
+
+* ``c_div()`` --- for integer ceiling divide, see
+  `relevant discussion thread <https://discuss.python.org/t/91269>`_.
+* ``gcdext()`` --- to solve linear `Diophantine equation <https://en.wikipedia.org/wiki/Diophantine_equation>`_ in two variables (the
+  :external+py3.14:class:`int` implementation actually includes an extended
+  Euclidean algorithm)
+* ``isqrt_rem()`` --- to return both integer square root and a remainder (if
+  integer isn't a perfect square)
+* ``ilog()`` --- integer logarithm, :external+py3.14:func:`math.log`
```

Comment:
@serhiy-storchaka proposed ilog2() in https://github.com/python/cpython/issues/81313. It may be more useful and common to need ilog2()?

Reply:
Isn't `ilog2(x)`, `x.bit_length() - 1`?  I agree `ilog2` is easier to understand.

But more importantly, why are we debating the contents of this section?  Anything added would probably need another PEP, right?

Reply:
I would say that all current special handling of integers in log-related functions is useful.  Though, as Tim Peters [said](https://github.com/python/cpython/issues/120950#issuecomment-2297051837) - implementation of an exact ilog() might not worth efforts.  (BTW, the SymPy has integer_log() function.)

But why not move to the new module "inexact" versions?  ilog10 and ilog2 will be helpful.  And then all functions in the math module will share same argument processing.